libs/application-generic/src/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts

```diff
@@ -121,68 +127,103 @@ export class GetSubscriberPreference {
       return acc;
     }, {});
 
-    const mergedPreferences: ISubscriberPreferenceResponse[] = workflowList.map(
-      (workflow) => {
-        const preferences = workflowPreferenceSets[workflow._id];
-        const mergeCommand = MergePreferencesCommand.create({
-          workflowResourcePreference: preferences.workflowResourcePreference,
-          workflowUserPreference: preferences.workflowUserPreference,
-          subscriberWorkflowPreference:
-            preferences.subscriberWorkflowPreference,
-          ...(subscriberGlobalPreference ? { subscriberGlobalPreference } : {}),
-        });
-        const merged = MergePreferences.execute(mergeCommand);
+    const workflowPreferences: ISubscriberPreferenceResponse[] =
+      this.calculateWorkflowPreferences(
+        workflowList,
+        workflowPreferenceSets,
+        subscriberGlobalPreference,
+        command.includeInactiveChannels,
+      );
+
+    const nonCriticalWorkflowPreferences = workflowPreferences.filter(
+      (preference) => !preference.template.critical,
+    );
+
+    return nonCriticalWorkflowPreferences;
+  }
```

Comment:
all I did in this file is moved the code parts to the class functions to be able to instrument them and see the perf bottleneck